doc/source/train/doc_code/xgboost_quickstart.py

```diff
@@ -0,0 +1,114 @@
+# flake8: noqa
+# isort: skip_file
+
+# __xgboost_start__
+import xgboost
+from sklearn.datasets import load_iris
+from sklearn.model_selection import train_test_split
+
+# 1. Load your data as an `xgboost.DMatrix`.
+data = load_iris(as_frame=True)
+train_X, eval_X, train_y, eval_y = train_test_split(
+    data["data"], data["target"], test_size=0.2
+)
+
+dtrain = xgboost.DMatrix(train_X, label=train_y)
+deval = xgboost.DMatrix(eval_X, label=eval_y)
+
+# 2. Define your xgboost model training parameters.
+params = {
+    "tree_method": "approx",
+    "objective": "reg:squarederror",
+    "eta": 1e-4,
+    "subsample": 0.5,
+    "max_depth": 2,
+}
+
+# 3. Do non-distributed training.
+bst = xgboost.train(
+    params,
+    dtrain=dtrain,
+    evals=[(deval, "validation")],
+    num_boost_round=10,
+)
+# __xgboost_end__
+
+
+# __xgboost_ray_start__
+import xgboost
+from sklearn.datasets import load_iris
+
+import ray.train
+from ray.train.xgboost import XGBoostTrainer, RayTrainReportCallback
+
+# 1. Load your data as a Ray Data Dataset.
+data = load_iris(as_frame=True).frame
+dataset = ray.data.from_pandas(data)
+train_dataset, eval_dataset = dataset.train_test_split(test_size=0.2)
```

Comment:
Let's just host a bunch of csv/parquet files on s3 and use `read_parquet`.

Also, read separate files/folders rather than use train_test_split.